src/cascadia/TerminalSettingsModel/DefaultProfileUtils.cpp

```diff
@@ -20,14 +20,15 @@ winrt::Microsoft::Terminal::Settings::Model::Profile CreateDefaultProfile(const
     const winrt::guid profileGuid{ Microsoft::Console::Utils::CreateV5Uuid(TERMINAL_PROFILE_NAMESPACE_GUID,
                                                                            gsl::as_bytes(gsl::make_span(name))) };
 
-    auto newProfile = winrt::make<winrt::Microsoft::Terminal::Settings::Model::implementation::Profile>(profileGuid);
-    newProfile.Name(name);
+    auto newProfile = winrt::make_self<winrt::Microsoft::Terminal::Settings::Model::implementation::Profile>(profileGuid);

```

Comment:
Why do we need the implementation type here?

Reply:
I specifically only exposed the getter for `Origin` in the projected type, not the setter. So to set it, we need this to be the impl type.